### DISCUSSION TITLE
markdown as props.children in a custom component

### DISCUSSION THREAD

**Discussion by gabriele-ct** [mdx-js/mdx#2569] [Q&A] [answered]
Hi, 
In the previous version of my project I was using mdx v1 and when I had something like this

```
# some markdown

<CustomComponent>
This is **markdown** text
</CustomComponent>

## some other markup
```

In my CustomComponent implementation I was getting as `props.children` a string exacly matching the un processed markdown string, in this case `This is **markdown** text`

However in my current implementation which uses mdx v3, I get as `props.children` a processed mdx tree (with html tags replaced) . Is this something that has been implemented with the v3 version or am I doing something wrong?

Thank you

**Comment by ChristianMurphy** [comment, marked as answer]
That is markdown interleaving, it is intentional and has been the default for years https://github.com/mdx-js/mdx/issues/628
You can find documentation on the rules here: https://mdxjs.com/docs/what-is-mdx/#interleaving

**Comment by remcohaszing** [comment]
If you’re interested in the raw text content, you can use a JSX expression with a string literal.

```mdx
# some markdown

<CustomComponent>
{'This is **markdown** text'}
</CustomComponent>

## some other markup
```
